File: ios/Classes/Model/UserResponse.swift
//
//  UserResponse.swift
//  integration_test
//
//  Created by Najam Us Saqib on 26/02/2024.
//

import Foundation

// MARK: - User Mode -

struct User {
    let displayName: String?
    let avatarId: String?
    let avatarUrl: String?
    let externalId: String?
    let tokenId: String?
    let profileLink: String?
    
    func toMap() -> [String : Any] {
        return [
            "displayName" : displayName as Any,
            "avatarId" : avatarId as Any,
            "avatarUrl" : avatarUrl as Any,
            "externalId" : externalId as Any,
            "tokenId" : tokenId as Any,
            "profileLink" : profileLink as Any
        ]
    }
}

// MARK: - User Response -

struct UserResponse: SnapchatResponse {
    var code: Int = 200
    var message: String = ""
    var user: User?
    
    func toMap() -> [String : Any?] {
        return [
            "code" : code,
            "message" : message,
            "user" : user?.toMap() as Any
        ]
    }
}

File: lib/src/user_response.dart
import 'package:snapchat_loginkit/src/snapchat_response.dart';

class User {
  final String? displayName;
  final String? avatarUrl;
  final String? externalId;
  final String? tokenId;
  final String? avatarId;
  final String? profileLink;

  User(
      {this.displayName,
      this.avatarUrl,
      this.externalId,
      this.tokenId,
      this.avatarId,
      this.profileLink});

  factory User.fromMap(Map<Object?, Object?> map) => User(
      displayName:
          map['displayName'] != null ? map['displayName'] as String : null,
      avatarUrl: map["avatarUrl"] != null ? map["avatarUrl"] as String : null,
      externalId:
          map['externalId'] != null ? map['externalId'] as String : null,
      avatarId: map['avatarId'] != null ? map['avatarId'] as String : null,
      profileLink:
          map['profileLink'] != null ? map['profileLink'] as String : null,
      tokenId: map['tokenId'] != null ? map['tokenId'] as String : null);
}

class UserResponse extends SnapchatResponse {
  User? user;

  UserResponse({super.code, super.message, this.user});

  factory UserResponse.fromMap(Map<Object?, Object?> map) => UserResponse(
      code: map['code'] as int? ?? 0,
      message: map['message'] as String? ?? "",
      user: map['user'] != null
          ? User.fromMap(map['user'] as Map<Object?, Object?>)
          : null);
}
